### PR TITLE
remove merging label when merge is cancelled

### PR DESCRIPTION
**Pull request by PaliC**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #98967

Adds a script to get rid of the "merging" label when a job is cancelled. 
At the moment this can create a race condition is someone cancels a job and starts a new one, though these cases should be pretty rare especially in cases where its from a new merge command.

### CODE REVIEW

File: .github/scripts/on_cancel_merge.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+    repo = GitRepo(get_git_repo_dir(), get_git_remote_name(), debug=True)
+    org, project = repo.gh_owner_and_name()
+    pr_num = args.pr_num
+
+    GitHubPR(org, project, pr_num).remove_label(MERGE_IN_PROGRESS_LABEL)
```
Comment (malfet) [inline .github/scripts/on_cancel_merge.py:24]: Please add a new line
```suggestion
    GitHubPR(org, project, pr_num).remove_label(MERGE_IN_PROGRESS_LABEL)

```

**Review by malfet — APPROVED** [review]

File: .github/workflows/trymerge.yml [hunk 2]
```diff
@@ -75,6 +75,7 @@ jobs:
           PR_NUM: ${{ github.event.client_payload.pr_num }}
         run: |
           set -ex
+          python3 .github/scripts/on_cancel_merge.py "${PR_NUM}"
           python3 .github/scripts/comment_on_pr.py "${PR_NUM}" "merge"
```
Comment (malfet) [inline .github/workflows/trymerge.yml:79]: Can we merge those scripts together? (If `comment_on_pr.py` already imports GitHub class

Comment (PaliC) [reply .github/workflows/trymerge.yml:79]: I was originally going to do that, however, comment on pr is used in other bits which are not related to merging.

**Pull request closed without merging**